src/content-helper/dashboard-widget/provider.ts

```diff
@@ -0,0 +1,104 @@
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+import apiFetch from '@wordpress/api-fetch';
+
+/**
+ * Internal dependencies
+ */
+import {
+	ContentHelperError,
+	ContentHelperErrorCode,
+} from '../../blocks/content-helper/content-helper-error';
+import { TopPostData } from './top-posts/model';
+import {
+	convertDateToString,
+	removeDaysFromDate,
+} from '../../blocks/shared/utils/date';
+
+/**
+ * The form of the response returned by the /stats/posts WordPress REST API
+ * endpoint.
+ */
+interface TopPostsApiResponse {
+	error?: Error;
+	data?: TopPostData[];
+}
+
+/**
+ * The form of the result returned by the getTopPosts() function.
+ */
+export interface GetTopPostsResult {
+	posts: TopPostData[];
+}
+
+export const TOP_POSTS_DEFAULT_LIMIT = 3;
```

Comment:
I know the decision has been already taken to only show 3 posts but for me the numbers seems very very low. I think we should go with at least 5 but I am more inclined toward 7 or 10.

Reply:
This is as per the requirements [here](https://github.com/Parsely/wp-parsely/issues/1221#issuecomment-1353319380). I've asked about this myself in internal communications, and the decision was for it to remain 3. So for now this remains as-is.

Reply:
Since our main Goal here is to show users what Parse.ly is capable so I think more than 3 is better.

@danielabloch can you please give it another thought. Thanks

Reply:
This is on my list of things for us to revisit as well @mehmoodak and @acicovic but for now, let's keep it at 3.